api/grpc/server/server.go

```diff
@@ -56,6 +57,9 @@ func (s *apiServer) CreateContainer(ctx context.Context, c *types.CreateContaine
 	e.Runtime = c.Runtime
 	e.RuntimeArgs = c.RuntimeArgs
 	e.StartResponse = make(chan supervisor.StartResponse, 1)
+	if runt.GOOS == "solaris" && len(c.Id) >= 12 {
```

Comment:
ok, can you factor this logic out into a function to truncate the id?  its duplicated in a couple of the handlers